core/src/test/scala/com/mesosphere/usi/core/integration/SchedulerIntegrationTest.scala

```diff
@@ -110,4 +110,45 @@ class SchedulerIntegrationTest extends AkkaUnitTest with MesosClusterTest with I
       }
     }
   }
+
+  "terminates the stream when Mesos master dies" in {
+    Given("A Mesos client")
+    lazy val mesosClient: MesosClient = MesosClient(mesosClientSettings, frameworkInfo).runWith(Sink.head).futureValue
+    lazy val factory = SchedulerFactory(mesosClient, InMemoryPodRecordRepository(), schedulerSettings, DummyMetrics)
+    lazy val (_, schedulerFlow) = factory.newSchedulerFlow().futureValue
+    lazy val (input, output) = commandInputSource
+      .log("scheduler commands")
+      .via(schedulerFlow)
+      .log("scheduler events")
+      .toMat(Sink.queue())(Keep.both)
+      .withAttributes(Attributes
+        .logLevels(onElement = Logging.DebugLevel, onFinish = Logging.InfoLevel, onFailure = Logging.ErrorLevel))
+      .run
+
+    And("a first successful command.")
+    input.offer(commands.KillPod(PodId("unknown-pod-1"))).futureValue
+    inside(output.pull().futureValue.value) {
+      case PodSpecUpdatedEvent(PodId(idString), _) =>
+        idString should be("unknown-pod-1")
+    }
+
+    When("Mesos masters crash.")
+    logger.info("Kill Mesos masters.")
+    mesosCluster.masters.foreach(_.stop())
+
+    Then("The stream terminates")
+    val completion = input.watchCompletion().futureValue
```

Comment:
The logs state
```
[scheduler commands] Downstream finished, cause: EntityStreamException: Entity stream truncatio    582 n. The HTTP parser was receiving an entity when the underlying connection was closed unexpectedly.
```
But the `watchComplete` returns a successful future with `Done`. Why is that, @timcharper?